yulya/step3/most-probable-letters.js

```diff
@@ -0,0 +1,108 @@
+const fs = require('fs');
+const readline = require('readline');
+
+if (process.argv.length < 4) {
+  console.error('Example: node statistics-counter.js filename.txt letter');
+  process.exit(1);
+}
+
+const filePath = process.argv[2];
+const letter = process.argv[3];
+
+if (letter.length !== 1 || !letter.match(/[a-z]/i)) {
+  console.error('The letter argument should be a single alphabet character.');
+  process.exit(1);
+}
+
+const dictionary = new Map();
+
+function createDictionary (line, dictionary) {
+  line = line.toLowerCase().normalize('NFKC').replace(/\n/g, ' ').replace(/\r/g, '').replace(/\s+/g, ' ').replace(/[^a-z ]/g, '');
```

Comment:
nit: `replace(/\n/g, ' ').replace(/\r/g, '')` can be `replace(/\r?\n/g, ' ')`. Reasoning: `\r` mostly always goes before `\n`. More details: https://en.wikipedia.org/wiki/Newline#History

However, actually, instead of string manipulation (that is quite expensive as it _always_ create a new string), you could skip characters in lines 21-37